File: OCR-Translate/Models/AIProvider.swift
import Foundation

enum AIProvider: String, CaseIterable, Equatable {
    case deepseek
    case openai
    case gemini

    var displayName: String {
        switch self {
        case .deepseek: "DeepSeek"
        case .openai:   "ChatGPT (OpenAI)"
        case .gemini:   "Gemini (Google)"
        }
    }

    var shortName: String {
        switch self {
        case .deepseek: "DeepSeek"
        case .openai:   "OpenAI"
        case .gemini:   "Gemini"
        }
    }

    var endpoint: String {
        switch self {
        case .deepseek: "https://api.deepseek.com/v1/chat/completions"
        case .openai:   "https://api.openai.com/v1/chat/completions"
        case .gemini:   "https://generativelanguage.googleapis.com/v1beta/models/gemini-2.0-flash:generateContent"
        }
    }

    var model: String {
        switch self {
        case .deepseek: "deepseek-chat"
        case .openai:   "gpt-4o"
        case .gemini:   "gemini-2.0-flash"
        }
    }

    /// How the API key is sent
    var authHeader: (String, String)? {
        switch self {
        case .deepseek, .openai:
            return ("Authorization", "Bearer <key>")
        case .gemini:
            return nil // key goes in URL query param
        }
    }

    var getKeyURL: String {
        switch self {
        case .deepseek: "platform.deepseek.com → API Keys"
        case .openai:   "platform.openai.com → API Keys"
        case .gemini:   "aistudio.google.com → Get API Key"
        }
    }

    /// Key prefix hint for placeholder
    var keyPrefix: String {
        switch self {
        case .deepseek: "sk-"
        case .openai:   "sk-"
        case .gemini:   "AIza"
        }
    }
}

File: win/NGMproofread.Windows/Models/AIProvider.cs
namespace NGMproofread.Windows.Models;

public enum AIProvider
{
    DeepSeek,
    OpenAI,
    Gemini
}

public static class AIProviderExtensions
{
    public static string DisplayName(this AIProvider provider) => provider switch
    {
        AIProvider.DeepSeek => "DeepSeek",
        AIProvider.OpenAI => "ChatGPT (OpenAI)",
        AIProvider.Gemini => "Gemini (Google)",
        _ => ""
    };

    public static string ShortName(this AIProvider provider) => provider switch
    {
        AIProvider.DeepSeek => "DeepSeek",
        AIProvider.OpenAI => "OpenAI",
        AIProvider.Gemini => "Gemini",
        _ => ""
    };

    public static string Endpoint(this AIProvider provider) => provider switch
    {
        AIProvider.DeepSeek => "https://api.deepseek.com/v1/chat/completions",
        AIProvider.OpenAI => "https://api.openai.com/v1/chat/completions",
        AIProvider.Gemini => "https://generativelanguage.googleapis.com/v1beta/models/gemini-2.0-flash:generateContent",
        _ => ""
    };

    public static string Model(this AIProvider provider) => provider switch
    {
        AIProvider.DeepSeek => "deepseek-chat",
        AIProvider.OpenAI => "gpt-4o",
        AIProvider.Gemini => "gemini-2.0-flash",
        _ => ""
    };

    public static string KeyPrefix(this AIProvider provider) => provider switch
    {
        AIProvider.DeepSeek => "sk-",
        AIProvider.OpenAI => "sk-",
        AIProvider.Gemini => "AIza",
        _ => ""
    };

    public static string GetKeyURL(this AIProvider provider) => provider switch
    {
        AIProvider.DeepSeek => "platform.deepseek.com -> API Keys",
        AIProvider.OpenAI => "platform.openai.com -> API Keys",
        AIProvider.Gemini => "aistudio.google.com -> Get API Key",
        _ => ""
    };
}
